waymo2argo/waymo_raw_data_to_argoverse.py

```diff
@@ -0,0 +1,441 @@
+        standardcam_SE3_waymocam = SE3(
+            rotation=standardcam_R_waymocam, translation=np.zeros(3)
+        )
+        egovehicle_SE3_waymocam = SE3(
+            rotation=egovehicle_SE3_waymocam[:3, :3],
+            translation=egovehicle_SE3_waymocam[:3, 3],
+        )
+        standardcam_SE3_egovehicle = standardcam_SE3_waymocam.right_multiply_with_se3(
+            egovehicle_SE3_waymocam.inverse()
+        )
+        egovehicle_SE3_standardcam = standardcam_SE3_egovehicle.inverse()
+        egovehicle_q_camera = rotmat2quat(egovehicle_SE3_standardcam.rotation)
+        x, y, z = egovehicle_SE3_standardcam.translation
+        qw, qx, qy, qz = egovehicle_q_camera
+        f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
+        cam_dict = {
+            "key": "image_raw_" + cam_name,
+            "value": {
+                "focal_length_x_px_": f_u,
+                "focal_length_y_px_": f_v,
+                "focal_center_x_px_": c_u,
+                "focal_center_y_px_": c_v,
+                "skew_": 0,
+                "distortion_coefficients_": [0, 0, 0],
+                "vehicle_SE3_camera_": {
+                    "rotation": {"coefficients": [qw, qx, qy, qz]},
+                    "translation": [x, y, z],
+                },
+            },
+        }
+        calib_dict["camera_data_"] += [cam_dict]
+    return calib_dict
+
+
+def dump_pose(
+    city_SE3_egovehicle: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves the SE3 transformation from city frame
+        to egovehicle frame at a particular timestamp
+
+    Args:
+        city_SE3_egovehicle: A (4,4) numpy array representing the
+                            SE3 transformation from city to egovehicle frame
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    x, y, z = city_SE3_egovehicle[:3, 3]
+    R = city_SE3_egovehicle[:3, :3]
+    assert np.allclose(city_SE3_egovehicle[3], np.array([0, 0, 0, 1]))
+    q = rotmat2quat(R)
+    w, x, y, z = q
+    pose_dict = {"rotation": [w, x, y, z], "translation": [x, y, z]}
+    json_fpath = f"{parent_path}/{log_id}/poses/city_SE3_egovehicle_{timestamp}.json"
+    check_mkdir(str(Path(json_fpath).parent))
+    save_json_dict(json_fpath, pose_dict)
+
+
+def dump_point_cloud(
+    points: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves point cloud as .ply file extracted from Waymo's range images
+
+    Args:
+        points: A (N,3) numpy array representing the point cloud created from lidar readings
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    points = points.astype(float)
```

Comment:
maybe we can add a small inline comment just mentioning that it must be float since integer arrays cause problems

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/4ce61dbfa76327be7e9627546897df849c7b8cff)